code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -596,8 +596,8 @@
 /**
   * Proc that handles a charge attack windup for a mob.
   */
-/mob/living/simple_animal/hostile/proc/enter_charge(var/atom/target)
-	if((mobility_flags & (MOBILITY_MOVE | MOBILITY_STAND)) != (MOBILITY_MOVE | MOBILITY_STAND) || charge_state)
+/mob/living/simple_animal/hostile/proc/enter_charge(atom/target)
+	if(charge_state || body_position == LYING_DOWN || HAS_TRAIT(src, TRAIT_IMMOBILIZED))
```

Comment:
another one that looks a bit odd, keep an eye on it.

Reply:
The old check was a bit weird and the author, when asked, didn't know what it was intended to check.
That's a bit of a pattern in the old system.
From what I understood, this was checking for reasons not to start a charge, such as already being charging and being stunned. The check would fail only when both were true, but you'd still be able to change if you could not move but could stand up.
These edge cases rarely happen with simple mobs, but now they shouldn't happen at all.